packages/calypso-e2e/src/lib/blocks/image-block.ts

```diff
@@ -28,14 +27,16 @@ export class ImageBlock {
 	}
 
 	/**
-	 * Uplaods the target file at the supplied path to WPCOM.
-	 *
-	 * @param {string} filePath Path to the file on disk.
+	 * @returns The image element
```

Comment:
nit: I think the JSDoc should be

```
@returns {Promise<ElementHandle>} The image element.
```

Reply:
Should this be es-linted?

Reply:
https://github.com/Automattic/wp-calypso/pull/58516/commits/3c62fd2a0a05f844c2489b54d08b673514692e3b

Reply:
Ideally it should be eslinted, though on the other hand I assume we have rules for JSDoc already existing for other parts of the Calypso codebase and I assume they do not use this convention that `calypso-e2e` uses?

Reply:
Not sure I follow. It looks like `calypso-e2e` uses a [deprecated](https://eslint.org/docs/rules/require-jsdoc) [`require-jsdoc`](https://github.com/Automattic/wp-calypso/blob/c2a3f57266c1bb6123cd9b474491c31f42fca6a9/packages/calypso-e2e/.eslintrc.js#L8) rule with a configuration that doesn't seem to require adding the `@returns` type. The rest of wp-calypso seems to [not require return type either](https://github.com/Automattic/wp-calypso/blob/f23e39ed46e67d14673c92d92a8b6671fab8d727/packages/eslint-plugin-wpcalypso/lib/configs/recommended.js#L129). I guess this makes sense for TS files as the return type is always defined with the func declaration:
```ts
/**
 * @returns Welcome message
 * /
function howdy: Promise< string > {
  return 'Howdy!'
}
```

I guess it's OK to skip the `@returns` type part then? Otherwise, I'd definitely turn that rule on for consistency.

Reply:
You touched on what I was trying to get at - we should try to use the same basic JSDoc rule on all packages. I am not as familiar with this side of things and so was trying to gauge whether `calypso-e2e` is the one that's different.

Reply:
True that our TypeScript eslint rules requires the return type to be specified in the declaration, so perhaps we can discontinue requiring the `@return` statement in the JSDoc.